src/workerd/jsg/setup.h

```diff
@@ -246,7 +246,7 @@ class IsolateBase {
   // for the caller to ensure that the frame is kept alive.
   void popAsyncFrame();
 
-  std::deque<AsyncContextFrame*> asyncFrameStack;
+  kj::Vector<AsyncContextFrame*> asyncFrameStack;
```

Comment:
Optional cleanup: I realized this doesn't need to be a vector at all, since it's only modified in `Scope`. I think it could be a single pointer, or `Maybe<AsyncContextFrame&>`. In `Scope`'s constructor, you save the old pointer into a property of `Scope`, then change the pointer. In the destructor you change it back.

Reply:
Hmm, good point... except, I bet the before and after hooks are always called with the stack empty, since they'd be called directly from the microtask loop?

But it's not a big deal.

Reply:
Probably yeah. I'll check that later and change it up if I can prove that it's not a problem